src/main/java/com/synopsys/integration/detect/tool/detector/DetectorTool.java

```diff
@@ -173,25 +174,16 @@ public DetectorToolResult performDetectors(final File directory, final DetectorR
             }
         }
 
-        final DetectorToolResult detectorToolResult = new DetectorToolResult();
+        Map<CodeLocation, DetectCodeLocation> codeLocationMap = createCodeLocationMap(detectorEvaluations, directory);
 
-        detectorToolResult.rootDetectorEvaluationTree = Optional.of(rootEvaluation);
-
-        detectorToolResult.applicableDetectorTypes = applicable;
-
-        detectorToolResult.codeLocationMap = detectorEvaluations.stream()
-                                                 .filter(DetectorEvaluation::wasExtractionSuccessful)
-                                                 .map(it -> codeLocationConverter.toDetectCodeLocation(directory, it))
-                                                 .map(Map::entrySet)
-                                                 .flatMap(Collection::stream)
-                                                 .collect(Collectors.toMap(Map.Entry::getKey, Map.Entry::getValue));
-
-        detectorToolResult.bomToolCodeLocations = new ArrayList<>(detectorToolResult.codeLocationMap.values());
-
-        logger.info("");
-        final NameVersionDecision nameVersionDecision = detectorNameVersionHandler.finalDecision();
-        nameVersionDecision.printDescription(logger);
-        detectorToolResult.bomToolProjectNameVersion = nameVersionDecision.getChosenNameVersion();
+        final DetectorToolResult detectorToolResult = new DetectorToolResult(
+            detectorNameVersionHandler.finalDecision().getChosenNameVersion().get(),
```

Comment:
Why is this .get() but the other was not? Did we change the type? Also why do we take in null as one of the parameters?

Reply:
I made the constructor take in a NameVersion and then set the field to Optional.of(<supplied value>)
as for why we pass in null for failedDetectorTypes, there wasn't one provided in the original code.  I could change it to pass in an empty HashSet

Reply:
But what happens when the ChosenNameVersion is Empty() - won't there be an exception? You shouldn't do a get without an isPresent check or use orElse(null).

Can we just remove the field then?

Reply:
I commented on DetectorToolResult. Once you make those changes, use the below code to not throw an exception.
```suggestion
            detectorNameVersionHandler.finalDecision().getChosenNameVersion().orElse(null),
```